libraries/tock-cells/src/map_cell.rs

```diff
@@ -80,24 +156,66 @@ impl<T> MapCell<T> {
 
     /// Puts a value into the `MapCell`.
     pub fn put(&self, val: T) {
-        self.occupied.set(true);
-        unsafe {
-            ptr::write(self.val.get(), MaybeUninit::<T>::new(val));
-        }
+        // This will ensure the value as dropped
+        self.replace(val);
     }
 
     /// Replaces the contents of the `MapCell` with `val`. If the cell was not
     /// empty, the previous value is returned, otherwise `None` is returned.
-    pub fn replace(&self, val: T) -> Option<T> {
-        if self.is_none() {
-            self.put(val);
-            None
-        } else {
-            unsafe {
+    /// In the event the cell is currently borrowed, returns Err(AlreadyBorrowed)
+    pub fn try_replace(&self, val: T) -> Result<Option<T>, MapCellErr> {
```

Comment:
Can you add tests for `try_replace` as well?  Exercising all code paths

Reply:
I added a further test.